test/CodeGen/AVR/ctlz.ll

```diff
@@ -0,0 +1,10 @@
+; RUN: llc < %s -march=avr | FileCheck %s
+
+define i8 @count_leading_zeros(i8) unnamed_addr {
+; CHECK-LABEL: count_leading_zeros:
```

Comment:
What kind of tests do you think would be most useful? Since we are basically allowing LLVM to autogenerate code to do these operations the long way, any change to that LLVM code could potentially change the generated machine code.

For example, here's `ctpop`:

```asm
count_population:                       ; @count_population
; BB#0:                                 ; %entry-block
        mov     r25, r24
        lsr     r25
        andi    r25, 85
        sub     r24, r25
        mov     r25, r24
        andi    r25, 51
        lsr     r24
        lsr     r24
        andi    r24, 51
        add     r24, r25
        mov     r25, r24
        lsr     r25
        lsr     r25
        lsr     r25
        lsr     r25
        add     r25, r24
        andi    r25, 15
        mov     r24, r25
        ret
```

Once I know what's useful for this, I can try to extrapolate to the other `ct*` intrinsics.